docarray/array/storage/memory/getsetdel.py

```diff
@@ -24,22 +24,31 @@ def _del_docs_by_slice(self, _slice: slice):
         del self._data[_slice]
 
     def _del_doc_by_id(self, _id: str):
-        self._del_doc_by_offset(self._id2offset[_id])
+        if self._enable_offset2id:
+            self._del_doc_by_offset(self._id2offset[_id])
+        else:
+            raise ValueError("Id2offset is disabled")
```

Comment:
I think in the `DocumentArrayInMemory` this should not be applied. For the `InMemory` one the list-like behavior should always work